### PR TITLE
bump kube-network-policies to v0.3.0

### LINKED ISSUE #12: [Flaky] Services should create endpoints for unready pods - Netfilter Bug
It flakes on the network policy jobs https://testgrid.k8s.io/sig-network-gce#network-policies,%20google-gce

Analyzing one occurrence https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-network-policies/1782274345247182848

I can see DNS request on the trace but never a try to connect to the Service, that may indicate that something is dealing with the DNS request or response

### PR DESCRIPTION
**Pull request by aojea**
/kind flake
Fixes https://github.com/kubernetes-sigs/kube-network-policies/issues/12

```release-note
NONE
```

### CODE REVIEW

**Comment by aojea** [comment]
/assign @BenTheElder 

depends on https://github.com/kubernetes/k8s.io/pull/6916

**Comment by aojea** [comment]
/test pull-kubernetes-e2e-gce-network-policies

it was missing the node name on the manifest

**Comment by aojea** [comment]
/hold cancel

ping @ameukam

**Comment by rikatz** [comment]
/lgtm
Tks

**Comment by aojea** [comment]
hmm

> Test_ServiceClusterIPSelector
> {Failed;Failed;  === RUN   TestWatchStreamSeparation/separate_RPC_&_watch_cache_context_>_only_watch_cache_gets_bookmarks

/retest

File: cluster/addons/kube-network-policies/kube-network-policies.yaml [hunk 1]
```diff
@@ -45,6 +45,11 @@ spec:
         - mountPath: /lib/modules
           name: lib-modules
           readOnly: true
+        env:
+        - name: MY_NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
```
Comment (danwinship) [inline cluster/addons/kube-network-policies/kube-network-policies.yaml:52]: boo. you should make it a command-line option

Comment (aojea) [reply cluster/addons/kube-network-policies/kube-network-policies.yaml:52]: what is the benefit?

Comment (aojea) [reply cluster/addons/kube-network-policies/kube-network-policies.yaml:52]: independently, we can always add it , I just want to fix the flake

Comment (aojea) [reply cluster/addons/kube-network-policies/kube-network-policies.yaml:52]: I will move it to a flag https://github.com/kubernetes-sigs/kube-network-policies/issues/43

**Pull request merged**